uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -742,16 +738,19 @@ mod filters {
         let ffi_func = callable.ffi_rust_future_complete(ci);
         let call = format!("UniffiLib.INSTANCE.{ffi_func}(future, continuation)");
         let call = match callable.return_type() {
-            Some(Type::External {
-                kind: ExternalKind::DataClass,
-                name,
-                ..
-            }) => {
-                // Need to convert the RustBuffer from our package to the RustBuffer of the external package
-                let suffix = KotlinCodeOracle.class_name(ci, &name);
-                format!("{call}.let {{ RustBuffer{suffix}.create(it.capacity.toULong(), it.len.toULong(), it.data) }}")
+            Some(return_type) => {
+                match return_type {
+                    Type::Enum { name, .. } | Type::Record { name, .. }
```

Comment:
Maybe this could be better future-proofed by matching on `return_type.ffi_type()`.  The extra conversion can kick in if that matches `FfiType::RustBuffer(Some(metadata))`.